test/cli_test.js

```diff
@@ -21,6 +23,40 @@ describe('cli', function() {
     process.chdir(oldCwd);
   });
 
+  describe('build', function() {
+    let exitStub;
+
+    beforeEach(function() {
+      exitStub = sinon.stub(process, 'exit');
+    });
+
+    afterEach(function() {
+      rimraf.sync('dist');
+    });
+
+    context('on successful build', function() {
+      it('cleanups tmp files', function(done) {
+        const cleanup = sinon.spy(Builder.prototype, 'cleanup');
+
+        cli(['node', 'broccoli', 'build', 'dist']);
+
+        process.nextTick(() => {
+          chai.expect(cleanup).to.be.calledOnce;
+          done();
+        });
+      });
+
+      it('closes process on completion', function(done) {
+        cli(['node', 'broccoli', 'build', 'dist']);
+
+        process.nextTick(() => {
+          chai.expect(exitStub).to.be.calledWith(0);
```

Comment:
If this assertion fails, i believe this test wont fail with a useful error message, rather it will time out.
Remember in mocha, assertions are thrown exceptions. That means the `done` will never be hit...

We will need to address that before we can merge this.

One option would be:

```js
process.nextTick(() => {
  try {
    chai.expect(cleanup).to.be.calledOnce;
    done();
  } catch(e) {
    done(e); // fail the test with the given error
  }
});
```

Reply:
maybe this behavior is somehow related to node version. If `nextTick` would be handled synchronously this would not have to be an issue. 

I'm open to suggestions how to handle those tests. 

Easy solution is to refactor `lib/cli` module and cover it with unit tests. But there is quite some value in full integration tests for command line tools, so I'm not sure.